backends/p4tools/modules/testgen/targets/bmv2/test/p4-programs/up4.p4

```diff
@@ -736,7 +736,9 @@ control PreQosPipe(inout parsed_headers_t hdr, inout local_metadata_t local_meta
                 }
             }
         }
-        Routing.apply(hdr, local_meta, std_meta);
+        if (hdr.ipv4.isValid()) {
```

Comment:
Won't it be better if Routing control checks if the header is valid?
I would assume Acl control would need the same check too ...